acquire/utils.py

```diff
@@ -9,9 +10,91 @@
 import traceback
 from enum import Enum
 from pathlib import Path
-from typing import Dict, List, Optional
+from stat import S_IRGRP, S_IROTH, S_IRUSR
+from typing import Dict, List, Optional, Union
 
-from acquire.outputs import OUTPUTS
+import acquire.outputs
+from dissect.target.helpers.fsutil import TargetPath
+from dissect.util.stream import AlignedStream
+
+
+class VolatileAlignedStream(AlignedStream):
+    """A volatile streaming class to handle files that live in volatile filesystem. Such as procfs or sysfs.
+    Handeling various edge-cases and OSErrors encountered within these filesystems.
+
+    Set max_filesize to 0 or `None` to remove the maximum filesize limitation.
+
+    Args:
+        path: The path to the file to acquire a VolatileStream of
+        string_flags: String flags, such as 'r' or 'rb' to open the file with.
+        flags: Integer flags. such as `O_RDONLY` or `O_NONBLOCK` to open the file with.
+        max_filesize: Set the maximum file size for streaming. Set to 0 or `None` to remove the maximum.
+    """
+
+    def __init__(
+        self,
+        path: Union[str, TargetPath],
+        string_flags: str = "rb",
+        flags: int = os.O_RDONLY | os.O_NONBLOCK | os.O_NOATIME,
+        max_filesize=5242880,  # 5mb
+    ):
+
+        self.max_filesize = max_filesize
+        self.fh = open(path, string_flags)
+        self.fd = self.fh.fileno()
+
+        fcntl.fcntl(self.fd, fcntl.F_SETFL, flags)
+
+        self.pos = 0
+        self.buf = b""
+
+        super().__init__()
+
+    def tell(self) -> int:
+        return self.pos
+
+    def _read(self, offset: int, length: int) -> bytes:
+        prev_pos = None
+
+        while True:
+            if self.is_writeonly:
+                break
+
+            prev_pos = self.tell()
+            self.buf += os.read(self.fd, 4096)
+
+            self.pos = len(self.buf)
+
+            if not self.max_filesize:
```

Comment:
You check `if not self.max_filesize` but then proceed to check against the `max_filesize`? I think you meant to do this the other way around?